docs/contrib/future.rst

```diff
@@ -0,0 +1,74 @@
+==========
+__future__ features
+==========
+
+.. versionadded:: 0.12.0
+
+Importing from ``__future__`` allows you to add features to
+Hy that are not yet in the main language, due to perfomance issues
+or being harder to debug.
+
+
+.. _tailrec:
+
+
+``TailRec``
+===========
+
+The ``(import [__future__ [TailRec]])`` command
+gives programmers a simple way to use tail-call optimization
+(TCO) in their Hy code, with no need for trampoline or recur.
```

Comment:
It would be a bit helpful if you went into more detail on what *trampoline* and *recur* are referring to here.